CONTRIBUTING.md

```diff
@@ -104,20 +126,60 @@ The installed [`pre-commit`](https://pre-commit.com/) hooks will automatically r
 to match the project's style. Note that you will have to stage and commit again if anything was updated!  
 Most of the time, running pre-commit will automatically fix any issues that arise.
 
-
 ### Pyright
 
-For type-checking, run `pdm run pyright` (append `-w` to have it automatically re-check on every file change).
+For type-checking, run `uv run nox -s pyright` (append `-- -w` to have it automatically re-check on every file change).
 > [!NOTE]
 > If you're using VSCode and pylance, it will use the same type-checking settings, which generally means that you don't necessarily have to run `pyright` separately.  
 > However, since we use a specific version of `pyright` (which may not match pylance's version), there can be version differences which may lead to different results.
 
-
 ### Changelogs
 
 We use [towncrier](https://github.com/twisted/towncrier) for managing our changelogs. Each change is required to have at least one file in the [`changelog/`](changelog/README.rst) directory, unless it's a trivial change. There is more documentation in that directory on how to create a changelog entry.
 
-
 ### Documentation
+
 We use Sphinx to build the project's documentation, which includes [automatically generating](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) the API Reference from docstrings using the [NumPy style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).  
-To build the documentation locally, use `pdm run docs` and visit http://127.0.0.1:8009/ once built.
+To build the documentation locally, use `uv run nox -s docs` and visit <http://127.0.0.1:8009/> once built.
+
+## Migrating development environments
+
+### Migrating from `pdm` to `uv`
+
+We've recently migrated from `pdm` to [`uv`][uv] for dependency and environment management for local development, in hopes of a developer experience with less friction.
+
+1. delete `pdm.lock` and `.pdm-python`
+2. run `uv run noxfile.py -s dev`to recreate the .venv, refresh installation of pre-commit, and install all dependencies for the nox environments.
+
+### Migrating `pdm`'s scripts to `nox`
+
+Previously, we were using pdm's scripts functions as a simple caller for our nox scripts.
+With our migration to uv, we have dropped a separate task caller entirely and switch to nox for all tasks.
```

Comment:
*Maybe* we could use pyprojects's [`[project.scripts]`](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#creating-executable-scripts) to shorten those commands?

Reply:
Ah right, it's public interface.